tmt/libraries/beakerlib.py

```diff
@@ -158,29 +159,42 @@ def fmf_node_path(self) -> Path:
         return super().fmf_node_path
 
     @property
-    def _library_cache(self) -> Dict[str, 'BeakerLib']:
+    def _library_cache(self) -> Dict[Path, 'BeakerLib']:
         # Initialize library cache (indexed by the repository name)
         # FIXME: cast() - https://github.com/teemtee/tmt/issues/1372
         if not hasattr(self.parent, '_library_cache'):
             cast(CommonWithLibraryCache, self.parent)._library_cache = dict()
 
         return cast(CommonWithLibraryCache, self.parent)._library_cache
 
+    @property
+    def _nonexistent_url(self) -> Set[str]:
+        # Set of url we tried to clone but didn't succeed
+        if not hasattr(self.parent, '_nonexistent_url'):
+            cast(CommonWithLibraryCache, self.parent)._nonexistent_url = set()
+
+        return cast(CommonWithLibraryCache, self.parent)._nonexistent_url
+
     def fetch(self) -> None:
         """ Fetch the library (unless already fetched) """
         # Check if the library was already fetched
         try:
-            library = self._library_cache[str(self.repo)]
+            library = self._library_cache[self.repo]
```

Comment:
I'm not sure whether it could happen, but, not all `Path` instances pointing to the same path are the same:

```python
>>> from tmt.utils import Path
>>> Path('/etc') == Path('/etc/')
True
>>> Path('/etc') == Path('/etc/../etc')
False
>>> Path('/etc').resolve() == Path('/etc/../etc').resolve()
True
>>> Path('/etc').resolve()
Path('/etc')
>>> Path('/etc/../etc').resolve()
Path('/etc')
>>> 
```

Using them as they are as keys might be questionable, on the other hand, `str(self.repo)` would also result in two different keys, `"/etc/"` and `"/etc/../etc"`, and nobody complained so far... But, I would propose adding `.resolve()` call when using `Path` instances as mapping keys.

(And with that article about treating Python as a Rust on my mind, it might be a good idea to add some sort of `ResolvedPath` type, which would be always resolve, and use that one in mappings like this, never allowing a bare `Path` to be used. But that's a different patch :)

Reply:
I've removed `Use Path directly in _library_cache` commit and rebased to the latest main.